api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/RequestPaymentMessageProcessor.java

```diff
@@ -22,13 +39,49 @@
 @Slf4j
 public class RequestPaymentMessageProcessor implements MessageTypeProcessor {
 
+    private PaymentService paymentService;
+    private MessageContextLoader messageContextLoader;
+    private MessageQueueClient messageQueueClient;
+    private PaymentCycleService paymentCycleService;
+    private EventAuditor eventAuditor;
+
     @Override
     @Transactional(Transactional.TxType.REQUIRES_NEW)
     public MessageStatus processMessage(Message message) {
-        // TODO AFHS-405 add processing logic when implementing
+        RequestPaymentMessageContext messageContext = messageContextLoader.loadRequestPaymentMessageContext(message);
+        PaymentCycle paymentCycle = messageContext.getPaymentCycle();
+        PaymentType paymentType = messageContext.getPaymentType();
+
+        if (paymentType == PaymentType.FIRST_PAYMENT) {
+            PaymentCalculation paymentCalculation = aFullPaymentCalculation(paymentCycle.getTotalEntitlementAmountInPence());
+            makePayment(paymentCycle, INITIAL_PAYMENT, paymentCalculation, paymentType);
+        } else {
+            PaymentCalculation paymentCalculation = paymentService.calculatePaymentAmount(paymentCycle);
+            paymentCycleService.updatePaymentCycleCardBalance(paymentCycle, paymentCalculation.getAvailableBalanceInPence());
```

Comment:
As discussed, the update of the payment cycle should happen in the complete message processor (in case of errors thrown by the DB), not here (which is what the above comment about a subtask refers to). I have changes that will achieve this - will discuss merging them into this PR in the morning (see attached patch).
[UpdatePaymentCycleInCompleteMessageProcessor.patch.txt](https://github.com/DepartmentOfHealth-htbhf/htbhf-claimant-service/files/4083051/UpdatePaymentCycleInCompleteMessageProcessor.patch.txt)

Reply:
PR has been updated so that the balance is updated on complete payment.